Update _index.md (#1835)
cncf/glossary: content/en/contribute/_index.md

    weight: 10
---

## Welcome 

Welcome to the Cloud Native Glossary contributing guide and thank you for your interest. If you would like to contribute to our project, checkout the following list of contribution types and their corresponding sections in this guide: 

1) [Work on an existing issue](#work-on-an-existing-issue)
2) [Propose new terms](#propose-new-terms)
3) [Update existing ones](#update-an-existing-term)
4) [Localize the glossary](#help-translate-the-glossary)

## CNCF glossary overview 

The goal of this glossary is to simplify the cloud native space — which is notorious for its complexity — and thus make it more accessible to people. 

The Cloud Native Glossary content is stored in [this GitHub repo](https://github.com/cncf/glossary) 
where you'll find a list of [issues](https://github.com/cncf/glossary/issues), pull requests ([PRs](https://github.com/cncf/glossary/pulls)), and 
[discussions](https://github.com/cncf/glossary/discussions) about the glossary. 

## Who can contribute?

> Basically, anyone can contribute, however, how you participate in the project depends on your level of expertise of the cloud native space. 

Reviewer: ```suggestion
How you can participate in this project depends on your level of cloud native expertise. 
```
I really want to stay away from "anyone can contribute." First of all, that is not true. We've seen a bunch of contributions from people who do not have the qualification to do so. While localization is an option, it isn't for people who only are proficient in English. 

We introduced this section because of the false belief that everyone can participate. Without the proper know-how, it creates a lot of frustration for contributors and maintainers. I want us to be very transparent here to avoid that. Currently, I'm pointing everyone who wants to contribute a new term to this section and have them confirm that they qualify. It may sound harsh, but we need to set clear expectations.

> Simplifying complex concepts requires a deep knowledge of the topic. Therefore, to contribute with a new term you must be proficient in the cloud native space. 

Reviewer: ```suggestion
Simplifying complex concepts requires a deep knowledge of the topic. 
Therefore, to contribute new terms, you must be proficient in them. 
Contributors are typically engineers who have worked with these technologies for some time or academics focused on cloud native. 

That know-how is required because explaining complex concepts in simple words is _really_ hard. And while the digestible, user-friendly outcome may seem easy, achieving the desired simplicity results from hard work and collaboration between cloud native experts. 
```

Moving that part up again to further explain that, just because the result looks easy, it meant it is easy.

> The localization effort is where beginners can make valuable contributions to the glossary. 

Reviewer: ```suggestion
The localization effort is where beginners proficient in another language can make valuable contributions to the Glossary. 
```

> With solid existing definitions in English, less experienced contributors can localize terms to a target language. You can join an existing localization team or create a new one. Read the section [Help translate the glossary](#help-translate-the-glossary) of this guide to learn how to get started. 

Reviewer: ```suggestion
With solid existing definitions in English, less experienced contributors can localize terms to a target language. You can join an existing localization team or create a new one. Read this guide's [Help Localize the glossary](#help-localize-the-glossary) section to learn how to get started. 
```

> ## Before you start

Reviewer: Not sure how to do this, but external links should open in a new tab. @seokho-son @iamNoah1, can you help?

Reviewer 2: 'shift + click on the link' ? :)

Reviewer 3: FYI we no longer force people to open external links in new tabs. I wrote up some reasons for that which I’ll copy below:
As part of the accessibility improvements we’ve been working on for our websites, we’d like to get away from forcing external sites to open in new browser tabs. The idea behind this was originally to prevent people from leaving our site or at least giving them an easy way to get back to our site by leaving the tab open. This practice, however, is not recommended for a few reasons:
- It can be disorienting to screen reader users, those with cognitive impairments or people who magnify their screen
- In a lot of cases it actually makes it harder to go back to the page you left since the Back button won’t take you there. This affects all users and is exacerbated on mobile where the old window is never visible.
- Opening a new tab takes control out of the user’s hands
[See this article for a full analysis](https://www.nngroup.com/articles/new-browser-windows-and-tabs/).

Reviewer 4: Just FYI:
we cannot achieve this (make the link open in new tab) with markdown, so we have to use HTML if we want to.

Ref: https://stackoverflow.com/questions/4425198/can-i-create-links-with-target-blank-in-markdown

> Before making any contribution to the glossary, please make sure of the following:

Reviewer: ```suggestion
Before beginning your Glossary contributor journey, be sure to:
```

Reviewer: Ok :)

> 1. If you don't already have one, create a [GitHub account](https://docs.github.com/en/get-started/signing-up-for-github/signing-up-for-a-new-github-account). 

Reviewer: ```suggestion
1. Create a [GitHub account](https://docs.github.com/en/get-started/signing-up-for-github/signing-up-for-a-new-github-account), if you don't have one already. 
```

2. Sign the [Contributor License Agreement](https://docs.linuxfoundation.org/lfx/easycla/v2-current/contributors) (CLA). 
3. [Verify your commit signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)
4. Enable [vigilant mode](https://docs.github.com/en/authentication/managing-commit-signature-verification/displaying-verification-statuses-for-all-of-your-commits#about-vigilant-mode) in your GitHub account to display the "Verified" status on your commits. 

## Best practices {#best-practices}

> To facilitate the reviewing process, our project uses [semantic line breaks](https://sembr.org/) (e.g., one line per sentence).

Reviewer: ```suggestion
To facilitate the reviewing process, please use [semantic line breaks](https://sembr.org/) (e.g., one line per sentence).
```

We recommend checking out this [markdown cheat sheet](https://www.markdownguide.org/cheat-sheet/) 
to correctly format Markdown text in GitHub (e.g., hyperlink, bold, italic).

> To name the .md file of each term we use lowercase letters, hyphens instead of spaces to separate words, and we don't use parenthesis.

Reviewer: ```suggestion
And when naming .md files, please use lowercase letters and hyphens instead of spaces to separate words and avoid parenthesis.
```

## Style guide

Read our [Style Guide](/style-guide/) to understand our guidelines for formatting and writing documents and make the contribution process more efficient. 

## Join the Glossary community! {#join-the-glossary-community}

If you want to contribute regularly, consider joining our monthly Glossary Working Group meetings. 
You can find the meeting details in the [CNCF calendar](https://www.cncf.io/calendar/). 
You can also connect with the maintainers and fellow contributors in our [#glossary](https://cloud-native.slack.com/archives/C02TX20MQBB) channel on the CNCF Slack 
— we'd love to meet you! 

## Work on an existing issue {#work-on-an-existing-issue}

> Go to the [Glossary GitHub repo issues](https://github.com/cncf/glossary/issues) where you will find a list of available issues. You can use labels (e.g., English language, help needed, good first issue) to filter out issues. 

Reviewer: ```suggestion
Go to the [Glossary GitHub repo issues](https://github.com/cncf/glossary/issues) to find a list of available issues. 
You can use labels (e.g., English language, help needed, good first issue) to filter out issues.
```

![Issue and labels](/images/how-to/issue-and-labels.png)

Be sure the term you select hasn't already been assigned to anybody. 
For example, here you can see that the first three terms are available while the fourth one has already been assigned.

![assigning a term](/images/how-to/howto-04.png)

> Once you select a term to work on, write a comment in the issue. 

Reviewer: ```suggestion
Once you select a term to work on, comment on the issue. 
```

![Claiming an issue](/images/how-to/claiming-an-issue.png)

Additionally, notify the maintainers on the [#glossary](https://cloud-native.slack.com/archives/C02TX20MQBB) channel of the CNCF Slack workspace and 

> tag _@Catherine Paganini_, _@jmo_, _@Seokho Son_, _@Jihoon Seo_, and/or _@iamnoah_ to be sure they don't miss it. 

Reviewer: ```suggestion
tag _@Catherine Paganini_, _@Seokho Son_, _@Jihoon Seo_, and/or _@iamnoah_ to be sure they don't miss it. 
```
Removing Jason :)

## CNCF glossary overview 

The goal of this glossary is to simplify the cloud native space — which is notorious for its complexity — and thus make it more accessible to people. 

The Cloud Native Glossary content is stored in [this GitHub repo](https://github.com/cncf/glossary) 
where you'll find a list of [issues](https://github.com/cncf/glossary/issues), pull requests ([PRs](https://github.com/cncf/glossary/pulls)), and 
[discussions](https://github.com/cncf/glossary/discussions) about the glossary. 

## Who can contribute?

How you can participate in this project depends on your level of cloud native expertise. 
Simplifying complex concepts requires a deep knowledge of the topic. 
Therefore, to contribute new terms, you must be proficient in them. 
Contributors are typically engineers who have worked with these technologies for some time or academics focused on cloud native. 

That know-how is required because explaining complex concepts in simple words is _really_ hard. And while the digestible, user-friendly outcome may seem easy, achieving the desired simplicity results from hard work and collaboration between cloud native experts. 

> 

Reviewer: ```suggestion

If you are not a cloud native expert yet but still want to contribute, we recommend teaming up with someone who is. 
Once the expert is confident that the term accurately describes the concept, you can submit a PR.
```

If you are not a cloud native expert yet but still want to contribute, we recommend teaming up with someone who is. 
Once the expert is confident that the term accurately describes the concept, you are ready for your first Glossary contribution.

The localization effort is where beginners proficient in another language can make valuable contributions to the Glossary. 
With solid existing definitions in English, less experienced contributors can localize terms to a target language. You can join an existing localization team or create a new one. Read this guide's [Help Localize the glossary](#help-localize-the-glossary) section to learn how to get started. 

## Before you start

Before beginning your Glossary contributor journey, be sure to complete the following steps:

1. Create a [GitHub account](https://docs.github.com/en/get-started/signing-up-for-github/signing-up-for-a-new-github-account), if you don't have one already. 
2. Sign the [Contributor License Agreement](https://docs.linuxfoundation.org/lfx/easycla/v2-current/contributors) (CLA). 

> 3. [Verify your commit signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)
> 4. Enable [vigilant mode](https://docs.github.com/en/authentication/managing-commit-signature-verification/displaying-verification-statuses-for-all-of-your-commits#about-vigilant-mode) in your GitHub account to display the "Verified" status on your commits. 

Reviewer 2: Hi @CathPag @valeriahhdez 
I'm not sure step 3 is an actionable step at this early stage. I believe that commit signature verification can only be done after a contributor has pushed a commit (or opened a PR) to GitHub repository. I think we can skip the step 3 and utilize the link only.

```suggestion
3. Enable [vigilant mode](https://docs.github.com/en/authentication/managing-commit-signature-verification/displaying-verification-statuses-for-all-of-your-commits#about-vigilant-mode) in your GitHub account to display the "[Verified](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)" status on your commits. 
```

Author: The link I provided in this step is to verify commit signatures with SSH or GPG keys, which can be completed using the GitHub interface before making any commit or push. Maybe you're referring to the -S flag used to verify with the git commit command?
We could add that option too and make it clear that signing commits this way is only required at the moment of committing the changes.

Reviewer 2: Hi @valeriahhdez 
Thank you for the clarification. I misunderstood. I thought the step is to check if a commit contains a verified signature. 
I think we can merge step 3, 4 into one since they are very related subjects (related with commit signature). It is to simplify the steps.

Applying my comment is up to you.
LGTM as is :) 

```suggestion
3. [Configure your commit signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) and enable [vigilant mode](https://docs.github.com/en/authentication/managing-commit-signature-verification/displaying-verification-statuses-for-all-of-your-commits#about-vigilant-mode) in your GitHub account to display the "Verified" status on your commits. 
```
